quill-core/src/main/scala/io/getquill/quotation/Parsing.scala

```diff
@@ -328,23 +330,107 @@ trait Parsing {
   private def identClean(x: Ident): Ident = x.copy(name = x.name.replace("$", ""))
   private def ident(x: TermName): Ident = identClean(Ident(x.decodedName.toString))
 
+  /**
+   * A simple 'transformer' that just searches a ast subtree for an instance of a conditional.
+   * this is needed in order to be able to detect if there are conditionals inside of optional statements
+   * since they could be nested in the body's subtree.
+   */
+  class SubtreeSearch(val state: Boolean) extends StatefulTransformer[Boolean] {
```

Comment:
Can you use https://github.com/getquill/quill/blob/master/quill-core/src/main/scala/io/getquill/ast/CollectAst.scala and check the list size?

Reply:
Ah, looks like I've re-implemented the wheel here. Will change.